modules/3d-tiles/src/lib/tileset/tile-3d-header.js

```diff
@@ -222,6 +222,12 @@ export default class Tile3DHeader {
     return this._header.extras;
   }
 
+  get hasChildren() {
+    // this.children are Tile3DHeader objects with content fetched from server
+    // this._header.children are children of this tile which are not yet fetched
+    return this.children.length > 0 || (this._header.children && this.children.length > 0);
```

Comment:
Can this just be `return this.children.length > 0;`

Reply:
it cannot work well with i3s tiles, the i3s tile doesn't include the tree structure in tileset, so loaders.gl needs to keep fetching each unfetched tile node during the traversal. So if an i3s tile node is already fetched `tile._header.children` will indicate if this tile has children or not. If loaders.gl continue fetching this tile's children, then whenever the child node is fetched, it will be constructed as Tile3DHeader object and added to tile.children.

Reply:
Maybe explain this in comment?